src/main/java/de/zalando/zmon/dataservice/data/MetricTiers.java

```diff
@@ -0,0 +1,98 @@
+package de.zalando.zmon.dataservice.data;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.collect.ImmutableMap;
+import de.zalando.zmon.dataservice.TokenWrapper;
+import de.zalando.zmon.dataservice.components.DefaultObjectMapper;
+import de.zalando.zmon.dataservice.config.DataServiceConfigProperties;
+import org.apache.http.client.fluent.Executor;
+import org.apache.http.client.fluent.Request;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.scheduling.annotation.Scheduled;
+import org.springframework.stereotype.Component;
+
+import java.io.IOException;
+import java.util.*;
+
+@Component
+public class MetricTiers {
+    public static final Logger LOG = LoggerFactory.getLogger(MetricTiers.class);
+
+    private final ObjectMapper objectMapper;
+    private final Executor executor;
+    private final TokenWrapper wrapper;
+    private final String hostname;
+
+    private Integer ingestMaxCheckTier;
+    private Set<Integer> criticalChecks;
+    private Set<Integer> importantChecks;
+
+    public MetricTiers(@DefaultObjectMapper ObjectMapper objectMapper,
+                       @SuppressWarnings("SpringJavaInjectionPointsAutowiringInspection") DataServiceConfigProperties config,
+                       @Qualifier("accessTokens") TokenWrapper wrapper) {
+        this.objectMapper = objectMapper;
+        this.hostname = config.getProxyControllerBaseUrl();
+        this.wrapper = wrapper;
+        this.executor = HttpClientFactory.getExecutor(
+                config.getRestMetricSocketTimeout(),
+                config.getRestMetricTimeout(),
+                config.getRestMetricConnections(),
+                config.getConnectionsTimeToLive()
+        );
+    }
+
+    @Scheduled(fixedRate = 60_000)
+    public void updateTiers() {
+        LOG.debug("Updating metric tiers configuration");
+        try {
+            this.ingestMaxCheckTier = getMaxCheckTier();
+
+            final Map<String, Set<Integer>> checkTiers = getCheckTiers();
+            this.criticalChecks = checkTiers.get("critical");
+            this.importantChecks = checkTiers.get("important");
+        } catch (IOException e) {
+            LOG.error("Metric tiers configuration update failed", e);
+        }
+    }
+
+    public boolean isMetricEnabled(int checkId) {
+        if (ingestMaxCheckTier <= 0 || ingestMaxCheckTier >= 3) return true;
+        else if (ingestMaxCheckTier == 1) return criticalChecks.contains(checkId);
+        else return criticalChecks.contains(checkId) || importantChecks.contains(checkId);
```

Comment:
minor, but feels like this should be a switch statement, because `ingestMaxCheckTier=0` and `ingestMaxCheckTier=3` are quite different cases

Reply:
it's the same case, 0 and 3 means "everything is allowed", and they both are handled by row 62